Documentation/leonardo.rst

```diff
@@ -0,0 +1,247 @@
+Leonardo
+========
+
+Website: https://leonardo-supercomputer.cineca.eu/
+
+We will focus here on the Booster module of Leonardo, i.e. we use GPUs. That is, one node consists of:
+
+- 1× Intel Xeon Platinum 8358 (Sapphire Rapids) CPU @ 2 sockets × 32 cores × 2 HT
+- 4× Nvidia A100 SXM6 GPUs
+
+Thus, we will run SeisSol with 4 ranks per node. As architectures, we compile for the host/CPU architecture ``skx``, and use ``sm_80`` as architecture for the GPUs, together
+with CUDA as device backend. For the SYCL parts, we use AdaptiveCpp (formerly known as hipSYCL or Open SYCL).
+
+Build Instructions (without Spack)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here, we go for a build using gcc and AdaptiveCpp. We begin by setting up an environment. Firstly, choose a folder you want to install SeisSol to and navigate to it.
+Run ``pwd`` and copy the path there. Run the following script there.
+
+.. code-block:: bash
+
+    export SEISSOL_BASE=$(pwd)
+    export SEISSOL_PREFIX=$SEISSOL_BASE/local
+    export CMAKE_PREFIX_PATH=$SEISSOL_PREFIX:$CMAKE_PREFIX_PATH
+
+    mkdir -p $SEISSOL_PREFIX
+
+Next, we load the necessary modules for our SeisSol build.
+We load GCC explicitly, so that the environment variables for ``CC``, ``CXX``, etc. are set for us automatically.
+
+.. code-block:: bash
+
+    module load python/3.10.8--gcc--11.3.0
+    module load cuda/11.8
+    module load openmpi/4.1.4--gcc--11.3.0-cuda-11.8
+    module load gcc/11.3.0
+    module load ninja
+    module load hdf5/1.12.2--openmpi--4.1.4--gcc--11.3.0
+    module load netcdf-c/4.9.0--openmpi--4.1.4--gcc--11.3.0
+
+It can be useful to place these module loads into a script of their own for running jobs later-on.
+
+Unforunately, we need to compile LLVM ourselves, as the given LLVM module does not have an NVPTX backend enabled (note: consider switching your build directory to TMP or SCRATCH, since LLVM can be big).
+
+.. code-block:: bash
+
+    export GCC_ROOT=$(dirname $(dirname $(which gcc)))
+    export CUDA_ROOT=$(dirname $(dirname $(which nvcc)))
+
+    wget https://github.com/llvm/llvm-project/archive/refs/tags/llvmorg-16.0.6.tar.gz
+    tar -xf llvmorg-16.0.6.tar.gz
+    mkdir -p llvm-project-llvmorg-16.0.6/build
+    cd llvm-project-llvmorg-16.0.6/build
+    cmake ../llvm -GNinja -DCMAKE_INSTALL_PREFIX=$SEISSOL_PREFIX -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;compiler-rt;openmp;polly" -DGCC_INSTALL_PREFIX="${GCC_ROOT}" -DCUDA_TOOLKIT_ROOT_DIR="${CUDA_ROOT}" -DLLVM_TARGETS_TO_BUILD="X86;NVPTX"
```

Comment:
You can also use to `lld` linker for building `llvm`. It is usually faster. Just add the following cmake option `-DLLD_BUILD_TOOLS=ON`. But, yes, you must be sure that `lld` is installed at the first place.

Reply:
Hi, thanks! I'll add it in the next update for the instructions (maybe we can also avoid re-building LLVM entirely somehow...).